### ISSUE TITLE
feat: better diagnostics when instance does not satisfy `abc.collections.<Class>`

### ISSUE THREAD

**Issue by NickCrews** [astral-sh/ty#2390]
I am working in the ibis project, trying to fix up its typing so that [it's `Schema` class](https://github.com/ibis-project/ibis/blob/e5921c9a851afc8fd3980f6a0bbfd73720b4d5d6/ibis/expr/schema.py#L25) typechecks against a `collections.abc.Mapping[str, ibis.DataType]`. It currently does not.

You can reproduce this with `uv pip install ibis-framework` and then

```python
from collections.abc import Mapping

import ibis

s: Mapping[str, ibis.DataType] = ibis.schema({"a": "int64", "b": "string"})
```

This gives the type error ``Object of type `Schema` is not assignable to `Mapping[str, DataType]`: Incompatible value of type `Schema`ty[invalid-assignment](https://ty.dev/rules#invalid-assignment)``

It would be great if this error message instead told me why it isn't assignable. What required methods does it not implement? Is it that the methods have the wrong signatures? Or gave some more actionable advice, like I need to make ibis.Schema inherit from collections.abc.Mapping or call `collections.abc.Mapping.register(ibis.Schema)`

Thank you!

**Comment by AlexWaygood** [comment]
Thanks!

**Comment by sharkdp** [comment]
I'm going to reopen this, because it seems like a distinct use case that we should consider when implementing #163 (I'm adding #163 as a parent). It's currently not clear to me how a more generic version of this check would look like, since we're not checking against a protocol, but rather against an ABC.

**Comment by carljm** [comment]
I'm not clear if the right course here is "give more detailed feedback about all nominal-instance subtyping failures" (e.g. give the MRO as an info diagnostic?) or "give more detailed advice about specific non-protocols that people might assume are protocols" (e.g. ABCs) or "specifically mention that type checkers ignore ABC `.register` calls for subtyping purposes", or some combination.

I'm not sure this issue is high priority for stable release -- I'm triaging it outside of the stable milestone for now.
